theseus/utils/examples/bundle_adjustment/util.py

```diff
@@ -0,0 +1,57 @@
+import math
+from typing import Tuple
+
+import torch
+
+
+def soft_loss_cauchy(
+    x: torch.Tensor, radius: torch.Tensor
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    ratio = (x + radius) / radius
+    val = torch.log(ratio) * radius
+    der = 1.0 / ratio
+    return val, der
+
+
+def soft_loss_huber_like(
+    x: torch.Tensor, radius: torch.Tensor
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    ratio = (x + radius) / radius
+    sq = torch.sqrt(ratio)
+    val = (sq - 1) * radius
+    der = 0.5 / sq
+    return val, der
+
+
+# returns a uniformly random point of the 2-sphere
+def random_s2(dtype: torch.dtype = torch.float64) -> torch.Tensor:
```

Comment:
Do we need these or can we replace with Taosha's inside the class version?

Reply:
@maurimo, can you check [this](https://github.com/facebookresearch/theseus/blob/main/theseus/geometry/so3.py#L46) and see if the sampling functions you added are complementary to those. If not, then it's better to replace with `th.rand_so3` or `th.randn_so3`.